cli/clone.ts

```diff
@@ -0,0 +1,130 @@
+  }
+
+  const codeAlreadyExists = fs.existsSync(path.join(process.cwd(), 'pack.ts'));
+  if (codeAlreadyExists) {
+    const shouldOverwrite = promptForInput('A pack.ts file already exists. Do you want to overwrite it? (y/N)?');
+    if (!shouldOverwrite.toLocaleLowerCase().startsWith('y')) {
+      return printAndExit('Aborting');
+    }
+  }
+
+  const client = createCodaClient(apiKey, formattedEndpoint);
+
+  let packVersion: string;
+  try {
+    const maybeVersion = await getPackLatestVersion(client, packId);
+    if (!maybeVersion) {
+      return printAndExit(`No built versions found for pack ${packId}. Only built versions can be cloned.`);
+    }
+    packVersion = maybeVersion;
+  } catch (err: any) {
+    maybeHandleClientError(err);
+    throw err;
+  }
+
+  let sourceCode: string | null;
+  try {
+    sourceCode = await getPackSource(client, packId, packVersion);
+  } catch (err: any) {
+    maybeHandleClientError(err);
+    throw err;
+  }
+
+  if (!sourceCode) {
+    print(`Unable to download source for Pack version ${packVersion}. Packs built using the CLI can't be cloned.`);
+
+    const shouldInitializeWithoutDownload = promptForInput(
+      'Do you want to continue initializing with template starter code instead (y/N)?',
+    );
+    if (!shouldInitializeWithoutDownload.toLocaleLowerCase().startsWith('y')) {
+      return process.exit(1);
+    }
+
+    await handleInit();
+    storePackId(manifestDir, packId, codaApiEndpoint);
+    return;
+  }
+
+  print(`Fetched source at version ${packVersion}`);
+
+  await handleInit();
+  storePackId(manifestDir, packId, codaApiEndpoint);
+
+  fs.writeFileSync(path.join(manifestDir, 'pack.ts'), sourceCode);
+  printAndExit("Successfully updated pack.ts with the Pack's code!", 0);
+}
+
+function maybeHandleClientError(err: any) {
+  if (isResponseError(err)) {
+    switch (err.response.status) {
+      case 401:
+      case 403:
+      case 404:
+        return printAndExit("You don't have permission to edit this pack.");
+    }
+  }
+}
+
+async function getPackLatestVersion(client: Client, packId: number) {
+  const {items} = await client.listPackVersions(packId, {limit: 1});
+  if (!items || !items[0]) {
+    return null;
+  }
+  return items[0].packVersion;
+}
+
+async function getPackSource(client: Client, packId: number, version: string) {
+  const {files} = await client.getPackSourceCode(packId, version);
+
+  if (files.length !== 1 || !files[0].filename.endsWith('.ts')) {
+    return null;
+  }
+
+  const onlyFile = files[0];
+
+  // Fetch existing source code
+  const response = await fetch(onlyFile.url, {
+    headers: {
+      'Content-Type': 'application/javascript',
+      'User-Agent': 'Coda-Typescript-API-Client',
+    },
+  });
+  if (response.status >= 400) {
+    return printAndExit(`Error while fetching pack source code: ${response.statusText}`);
+  }
+  const sourceCode = await response.text();
+  return sourceCode;
```

Comment:
Option, you can just do `return response.text();` if you prefer. You can just return promises all the way up until something actually needs to `await` them.

Reply:
Done